app/handlers/sessions_create.rb

```diff
@@ -36,7 +36,8 @@ class SessionsCreate
   protected
 
   def setup
-    @data = OmniauthData.new(request.env['omniauth.auth'])
+    @data = OmniauthData.new(request.env['omniauth.auth']) \
+      rescue fatal_error(code: :invalid_omniauth_data)
```

Comment:
I was planning to handle this in the else block but sure we can do this instead.